### PR TITLE
fix(developer): hint text special characters

### LINKED ISSUE #10372: bug(developer): special key text in key hints does not render with the OSK Symbol font
From #10371.

> While subkeys were already being adequately styled, I recently realized that hints were not considering such shortcuts. For the record, Developer doesn't either:

image

### PR DESCRIPTION
**Pull request by mcdurdin**
Fixes #10372.

# User Testing

* **TEST_HINT_TEXT:** In the touch layout editor, add a special hint text to a key, such as `*Shift*`, and verify that the hint displays as the Shift icon, not the text.

### CODE REVIEW

File: developer/src/tike/xml/layoutbuilder/builder.js [hunk 1]
```diff
@@ -744,15 +745,27 @@ $(function() {
...
   }, {saveOnce: true});
 
-  builder.keyHintChange = function(val) {
-    const key = builder.selectedKey();
-    key.data('hint', val);
-    $('.hint', key).text(builder.inferKeyHintText(val, $(key).data('longpress'), $(key).data('flick'), $(key).data('multitap')));
+  builder.updateHint = function(key) {
+    const val = key.data('hint');
+    const hintElement = $('.hint', key);
+    hintElement.text(builder.inferKeyHintText(val, $(key).data('longpress'), $(key).data('flick'), $(key).data('multitap')));
```
Comment (darcywong00) [inline developer/src/tike/xml/layoutbuilder/builder.js:751]: Does `$(key).data('longpress')` fallback to the current default `.` if nothing is specified?

Comment (mcdurdin) [reply developer/src/tike/xml/layoutbuilder/builder.js:751]: `inferKeyHintText` does that.

https://github.com/keymanapp/keyman/blob/e6db899453607b6c4de1c72656dfae6a41c15962/developer/src/tike/xml/layoutbuilder/builder.js#L299-L306

**Review by darcywong00 — APPROVED** [review]
lgtm

**Comment by bharanidharanj** [comment]
# Test Results

* **TEST_HINT_TEXT (FAILED):** Tested with the attached PR build (Keyman Developer 17.0.282-beta-test-10927) on Windows 10 OS and here is my observation: 1. In the touch layout editor, I clicked the Shift key. 2. Added a text '*Test* in the Hint field. Noticed that the same text appeared on the Shift Key instead of displaying as a Shift icon. Seems to be an issue.

**Comment by mcdurdin** [comment]
Thanks for the test @bharanidharanj. The result is as expected, because `*Test*` is not a valid special identifier. You can see the valid identifiers in the Keycap Value field at the top right of the editor. `*Shift*` is an example that is a valid special identifier. Could you retest please?

@keymanapp-test-bot retest

**Comment by bharanidharanj** [comment]
> Thanks for the test @bharanidharanj. The result is as expected, because `*Test*` is not a valid special identifier. You can see the valid identifiers in the Keycap Value field at the top right of the editor. `*Shift*` is an example that is a valid special identifier. Could you retest please?
> 
> @keymanapp-test-bot retest

@mcdurdin Sure, I will do it.

**Comment by bharanidharanj** [comment]
# Test Results

* **TEST_HINT_TEXT (PASSED):** Retested this PR as per Marc's suggestion and here is my observation: 1. Opened a Custom project in the Keyman Developer. 2. Typed a Special character in the Hint field and verified that the special character appears as a Shift icon on the Key. Seems to be working as expected.

**Comment by keyman-server** [comment]
Changes in this pull request will be available for download in [Keyman version 17.0.282-beta](https://keyman.com/downloads/releases/beta/17.0.282)

**Pull request merged**